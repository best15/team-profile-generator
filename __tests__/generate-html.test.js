const Employee = require('../lib/employee');
const Manager = require('../lib/manager');
const Engineer = require('../lib/engineer');
const Intern = require('../lib/intern');

const generateHtml = require('../lib/generate-html.js');


describe("generate-html", () => {

    it("generates empty html page when no employee provided", () => {
        expect(generateHtml([])).toMatchSnapshot();
    });

    it("generates html with employee cards when employee detail is provided", () => {

        const empManager = new Manager("kshetri", "001", "kshetri@email.com", "04123456");
        const NewEngineer = new Engineer("sam", "002", "sam@email.com", "sam-git");
        const NewIntern = new Intern("sun", "004", "sun@email.com", "SWA");
        expect(generateHtml([empManager], [NewEngineer], [NewIntern])).toMatchSnapshot();
    });



});