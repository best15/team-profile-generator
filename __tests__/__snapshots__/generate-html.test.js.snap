// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-html generates empty html page when no employee provided 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <link
      rel=\\"stylesheet\\"
      href=\\"https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.5.0/css/bootstrap.min.css\\"
    />
    <title>My Team</title>
  </head>
  <body>
    <header class=\\"mb-5\\">
      <nav class=\\"navbar bg-secondary\\">
        <div class=\\"container-fluid justify-content-center\\">
          <h1 class=\\"mb-3 text-center text-light\\">My Team</h1>
        </div>
      </nav>
    </header>
    <main class=\\"container-fluid\\">
      <section class=\\"row justify-content-center\\"></section>
    </main>
  </body>
</html>
"
`;

exports[`generate-html generates html with employee cards when employee detail is provided 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\" />
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <link
      rel=\\"stylesheet\\"
      href=\\"https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.5.0/css/bootstrap.min.css\\"
    />
    <title>My Team</title>
  </head>
  <body>
    <header class=\\"mb-5\\">
      <nav class=\\"navbar bg-secondary\\">
        <div class=\\"container-fluid justify-content-center\\">
          <h1 class=\\"mb-3 text-center text-light\\">My Team</h1>
        </div>
      </nav>
    </header>
    <main class=\\"container-fluid\\">
      <section class=\\"row justify-content-center\\">
        <div class=\\"card col-sm-10 col-md-6 col-lg-2 mr-2 mb-3 p-0\\">
          <div class=\\"card-header bg-primary\\">
            <h4>kshetri</h4>
            <p>Manager</p>
          </div>
          <ul class=\\"list-group list-group-flush\\">
            <li class=\\"list-group-item\\">ID:001</li>
            <li class=\\"list-group-item\\">Email:kshetri@email.com</li>
            <li class=\\"list-group-item\\">OfficeNumber: 04123456</li>
          </ul>
        </div>

        <div class=\\"card col-sm-10 col-md-6 col-lg-2 mr-2 mb-3 p-0\\">
          <div class=\\"card-header bg-primary\\">
            <h4>sam</h4>
            <p>Engineer</p>
          </div>
          <ul class=\\"list-group list-group-flush\\">
            <li class=\\"list-group-item\\">ID:002</li>
            <li class=\\"list-group-item\\">Email:sam@email.com</li>
            <li class=\\"list-group-item\\">GitHub: sam-git</li>
          </ul>
        </div>

        <div class=\\"card col-sm-10 col-md-6 col-lg-2 mr-2 mb-3 p-0\\">
          <div class=\\"card-header bg-primary\\">
            <h4>sun</h4>
            <p>Intern</p>
          </div>
          <ul class=\\"list-group list-group-flush\\">
            <li class=\\"list-group-item\\">ID:004</li>
            <li class=\\"list-group-item\\">Email:sun@email.com</li>
            <li class=\\"list-group-item\\">School: SWA</li>
          </ul>
        </div>
      </section>
    </main>
  </body>
</html>
"
`;
